### ISSUE TITLE
Adding cancel_all

### ISSUE THREAD

**Issue by jakirkham** [dask/distributed#1087]
Would be great to have a function like `cancel`, which does `cancel_all`. Though maybe such a thing exists it and I have missed in my reading of the API.

**Comment by mrocklin** [comment]
What would the semantics of cancel_all be?

**Comment by jakirkham** [comment]
Not sure if I understand the question, but something like `c.cancel_all()` where `c` is the `Client`. Basically am looking for something that would cancel any running or pending tasks abruptly and dispose of all associated memory.

**Comment by mrocklin** [comment]
Is restart sufficient?

**Comment by martindurant** [comment]
Restarts kills all jobs anywhere, I think this is requesting to kill tasks that a particular client knows about. Perhaps restart of the client would be enough to do that.

**Comment by jakirkham** [comment]
Past experiences with `restart` in my configuration has not been great. Have experienced hangs when trying to shutdown the distributed cluster afterwards (this was a couple versions back, 1.16.1?). Would be good to understand why this is, but that is a separate issue.

Also am trying to identify a memory leak. My suspicion is restarting would merely confound the issue either by cleaning up the leaked memory or losing all knowledge of it. In either case we won't know what caused the leak.

For these reasons a cleanup method would be better. In the short term, something equivalent would be very helpful.

**Comment by pitrou** [comment]
I agree this sounds like a reasonable addition. @jakirkham, do you want to submit a PR?
